rescan.go

```diff
@@ -482,24 +444,28 @@ func rescan(chain ChainSource, options ...RescanOption) error {
 	//
 	// TODO(wilmer): refactor this and handleBlockDisconnected into their
 	// own methods.
-	handleBlockConnected := func(ntfn *blockntfns.Connected) error {
+	handleBlockConnected := func(ntfn *blockntfns.Connected, isRetry bool) error {
+		// If we have any blocks to retry and the notification doesn't
+		// correspond to the current block being retried, we'll queue it
+		// up to retry after.
+		newHeader := ntfn.Header()
```

Comment:
Instead of doing it this way, what about sending every connected blocks into a new go routine, that would handle them? That way we could continue consuming `block connected` notifications while this go routine is fetching filters and handling retries.

Reply:
Where would we access the internal state of the rescan? Wouldn't the caller always be looking at the filtered block?

Reply:
Yeah and the filtered block is only updated once the filter is fetched. If I'm understanding correctly, you're suggesting a goroutine that is sent blocks to fetch filters for. If we continue consuming notifications while the goroutine is doing its thing, the rescan could end up tracking blocks ahead of the goroutine if we're unable to fetch filters. What benefit does the goroutine provide that is not already achieved?

Reply:
I imagine it would be much easier to reason about and unit test, as the consumer of this ordered queue of blocks can be tested in isolation. It will also pave the way for doing batch filter fetching as this queue grows.

Reply:
Why does batch filter fetching need to be handled there? As is, the query API handles that given hints from the caller. 

Concurrency wise, it's much easier to have all this in a single execution context as you don't need to reason about serialization beyond receiving the new block connected/disconnected notifications.

Reply:
We don't necessarily need to have it being handled in a different goroutine, but I mean we wouldn't proceed until we have been able to fetch the filter. (I added a comment on the line in `handleBlockConnected`)

My rationale for doing this in a new goroutine is to not stall consumption of block connected notifications, but not sure if that matters.

Reply:
> My rationale for doing this in a new goroutine is to not stall consumption of block connected notifications, but not sure if that matters.

No need to due to the subscription being able to buffer notifications.